src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,152 @@
+
+        assertNotNull(registeredUser);
+        assertNotNull(registeredUser.getId());
+        assertEquals(LONGER_LOGIN, registeredUser.getLogin());
+        assertEquals(VALID_PASSWORD, registeredUser.getPassword());
+        assertEquals(VALID_AGE, registeredUser.getAge());
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User user = createUser(VALID_LOGIN, null, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_shortPassword_notOk() {
+        User user = createUser(VALID_LOGIN, SHORT_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_exactlyAtPasswordLength_ok() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertNotNull(registeredUser.getId());
+        assertEquals(VALID_LOGIN, registeredUser.getLogin());
+        assertEquals(VALID_PASSWORD, registeredUser.getPassword());
+        assertEquals(VALID_AGE, registeredUser.getAge());
+    }
+
+    @Test
+    public void register_longerPassword_ok() {
+        User user = createUser(VALID_LOGIN, LONGER_PASSWORD, VALID_AGE);
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertNotNull(registeredUser.getId());
+        assertEquals(VALID_LOGIN, registeredUser.getLogin());
+        assertEquals(LONGER_PASSWORD, registeredUser.getPassword());
+        assertEquals(VALID_AGE, registeredUser.getAge());
+    }
+
+    @Test
+    public void register_nullAge_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_underageUser_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, UNDERAGE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_atAgeBoundary_ok() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, EXACTLY_AT_AGE);
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertNotNull(registeredUser.getId());
+        assertEquals(VALID_LOGIN, registeredUser.getLogin());
+        assertEquals(VALID_PASSWORD, registeredUser.getPassword());
+        assertEquals(EXACTLY_AT_AGE, registeredUser.getAge());
+    }
+
+    @Test
+    public void register_existingLogin_notOk() {
+        User existingUser = createUser(EXISTING_LOGIN, VALID_PASSWORD, VALID_AGE);
+        registrationService.register(existingUser); // Register user to ensure it exists
```

Comment:
```suggestion
        storage.people.add(existingUser);
```

check common mistakes file

Reply:
not fixed